subsys/bluetooth/mesh/adv_ext.c

```diff
@@ -140,12 +142,13 @@ static int buf_send(struct net_buf *buf)
 		atomic_set_bit(adv.flags, ADV_FLAG_UPDATE_PARAMS);
 	}
 
-	adv.cb = BT_MESH_ADV(buf)->cb;
-	adv.cb_data = BT_MESH_ADV(buf)->cb_data;
+	adv.buf = buf;
 
 	err = adv_start(&adv_param, &start, &ad, 1, NULL, 0);
+
+	bt_mesh_adv_send_start(duration, err, BT_MESH_ADV(buf));
+
 	net_buf_unref(buf);
```

Comment:
Personally, I would put this unref in the while loop, where it was acquired (from `net_buf_get`), but up to you. @jhedberg may feel stronger about this.